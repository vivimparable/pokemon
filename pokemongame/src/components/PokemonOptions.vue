<template>
  <div class="options-container">

        <ul v-for="i in arrayHijo" :key="i.id">
            <li @click="isThisOne(i.id)" >{{i.name}}</li>  
        </ul>

  </div>
</template>

<script>
export default {

props:{
    arrayHijo:{
      typeof:Array,
      required:true
    },
    numberReal:{
      type:Number,
    },

    BooleanName:{
      type:Boolean,
      required:true
    },


  },methods:{

    
    isThisOne (value){
      let number = this.arrayHijo.filter(a=> a.id ==this.numberReal );
    if(value == this.numberReal ){
        this.$emit('changeValue', true)

    }else{
        this.$emit('changeValue', false, number)
    }
           
        
    }
  },mounted(){


  }
}
</script>

<style scoped>
ul {
    list-style-type: none;
}
li {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}
</style>