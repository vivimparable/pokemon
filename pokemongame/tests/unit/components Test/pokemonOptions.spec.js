import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { pokemonMock } from "@/mocks/pokemon.mock.js";


describe("Pokemon options", () => {
  let wrapper = shallowMount(PokemonOptions);

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        arrayHijo: pokemonMock,
        numberReal:1,
        BooleanName:true
      },
    });
  });

  test("snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Mostrar 4 opciones correctamante", () => {

    const [l1,l2,l3,l4] = wrapper.findAll('li');
    expect(l1.text()).toBe('charmander')
    expect(l2.text()).toBe('mew')
    expect(l3.text()).toBe('kakuna')
    expect(l4.text()).toBe('mewtwo')

  })

  test("Emit",   () => {
   const [l1,l2,l3,l4] = wrapper.findAll('li');

    l1.trigger('click');
    l2.trigger('click');
    l3.trigger('click');
    l4.trigger('click');
    expect(wrapper.emitted('changeValue').length).toBe(4);
    expect(wrapper.emitted('changeValue')[0]).toBeTruthy();
    expect(wrapper.emitted('changeValue')[1][0]).toBeFalsy();
    expect(wrapper.emitted('changeValue')[2][0]).toBeFalsy();
    expect(wrapper.emitted('changeValue')[3][0]).toBeFalsy(); 
  });

  test('repracticar',()=>{
    const [boton1,boton2,boton3,boton4] = wrapper.findAll('li');

    boton1.trigger('click');
    boton2.trigger('click');

    expect(wrapper.emitted('changeValue').length).toBe(2);
    expect(wrapper.emitted('changeValue')[0]).toBeTruthy();
    expect(wrapper.emitted('changeValue')[1][0]).toBeFalsy();

    
  })
});
