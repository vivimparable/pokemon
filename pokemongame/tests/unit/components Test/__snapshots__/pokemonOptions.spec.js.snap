// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon options snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>charmander</li>
  </ul>
  <ul>
    <li>mew</li>
  </ul>
  <ul>
    <li>kakuna</li>
  </ul>
  <ul>
    <li>mewtwo</li>
  </ul>
</div>
`;
